ledger/acctonline.go

```diff
@@ -853,13 +864,24 @@ func (ao *onlineAccounts) TopOnlineAccounts(rnd basics.Round, voteRnd basics.Rou
 			heap.Push(topHeap, data)
 		}
 
-		var res []*ledgercore.OnlineAccount
-		for topHeap.Len() > 0 && uint64(len(res)) < n {
+		for topHeap.Len() > 0 && uint64(len(topOnlineAccounts)) < n {
 			acct := heap.Pop(topHeap).(*ledgercore.OnlineAccount)
-			res = append(res, acct)
+			topOnlineAccounts = append(topOnlineAccounts, acct)
+		}
+
+		totalOnlineStake, err = ao.onlineTotals(rnd)
```

Comment:
Why not just return the invalidOnlineAccounts total and then let the caller make the call to onlineTotals() directly? It isolates the use cases better somewhat IMO

Reply:
Initially I did return the total invalidOnlineAccounts and then called OnlineTotals and subtracted it within `LoadTree`, but I found it a bit awkward, since the return values to `TopOnlineAccounts` were:
1. The top accounts online and `voteRnd`
2. The total stake that is online but not valid by `voteRnd`
I don't think this would make a lot of sense for someone trying to reason about this code without being very familiar with the context.
However, returning the top account online as well as the total online weight (both for `voteRnd`) seemed a bit more coherent us.